src/api/spec/models/kiwi/repository_spec.rb

```diff
@@ -0,0 +1,70 @@
+require 'rails_helper'
+require 'rantly/rspec_extensions'
+
+RSpec.describe Kiwi::Repository, type: :model do
+  pending "add some examples to (or delete) #{__FILE__}"
+
+  describe 'validations' do
+    context 'for source_path' do
+      it { is_expected.to validate_presence_of(:source_path) }
+
+      ['dir', 'iso', 'smb', 'this'].each do |protocol|
+        it "valid" do
+          property_of {
+            protocol + '://' + sized(range(1, 199)) { string(/./) }
+          }.check(3) { |string|
+            is_expected.to allow_value(string).for(:source_path)
+          }
+        end
+      end
+
+      ['ftp', 'http', 'https', 'plain'].each do |protocol|
+        it "valid" do
+          property_of {
+            # TODO: improve regular expression to generate the URI
```

Comment:
Done: https://trello.com/c/nt5sCi4d/400-src-api-spec-models-kiwi-repository-spec-rb-improve-regular-expression-to-generate-the-uri